cast/src/lib.rs

```diff
@@ -1050,6 +1050,28 @@ impl SimpleCast {
         Ok(format!("0x{}{}", "0".repeat(64 - num_hex.len()), num_hex))
     }
 
+    pub fn left_shift(value: &str, bits: &str) -> Result<U256> {
+        let value = Self::get_hex_or_dec(value);
+        let bits = Self::get_hex_or_dec(bits);
+
+        Ok(value.shl(bits))
+    }
+
+    pub fn right_shift(value: &str, bits: &str) -> Result<U256> {
+        let value = Self::get_hex_or_dec(value);
+        let bits = Self::get_hex_or_dec(bits);
+
+        Ok(value.shr(bits))
+    }
+
+    fn get_hex_or_dec(value: &str) -> U256 {
+        if value.starts_with("0x") {
+            U256::from_str_radix(value, 16).expect("Unable to convert hex to U256")
+        } else {
+            U256::from_str_radix(value, 10).expect("Unable to convert decimal to U256")
+        }
```

Comment:
Imo we should try to be smarter about this (e.g. if no `0x` but the string contains a-f then we assume hex) since other commands support hex with and without prefix.

Also noting here that the output base should be configurable. In the issue the flag `--output` was suggested but I think a better name is probably `--base-out <base>`. Should only support base 10 and base 16 (maybe using the values `10`/`16` and `dec`/`hex`). The output base should default to hex as this is default for all other commands as well.

Reply:
For the input base detection, would you recommend the usage of regex ?
Also, for the output should we suffix by 0x if it's an hex ?

Reply:
I'd check the other commands, I don't remember if `0x` is outputted or not (I don't think it is)

For input base detection you don't need regex, just:

- Strip `0x` if it's there. If it is, assume hex
- If it's not, try converting from base 10, I think, and if that fails, try converting from base 16
- If above step fails, fail

We could also add a `--base-in <base>` flag for maximum control in the case someone passes e.g. `90` but want's it interpreted as hex (this is still possible by prepending `0x` but might be too obscure?)

Reply:
I found some places where it was indeed outputted with a trailing "0x":

- https://github.com/foundry-rs/foundry/blob/master/cast/src/lib.rs#L103-L104
- https://github.com/foundry-rs/foundry/blob/master/cast/src/lib.rs#L982-L991
- https://github.com/foundry-rs/foundry/blob/master/cast/src/tx.rs#L234

On the other side, the U256 type `from_str` trait appear to assume that the base is always 16: https://github.com/paritytech/parity-common/blob/master/ethereum-types/src/uint.rs#L68

So I think that a --base-in would be useful, I hope that it wouldn't be too verbose. Now, we have two options:

- Make the --base-in require and there won't be any confusion, but could be annoying to type
- Don't make it required and throw an error that explains that the `--base-in` argument should be added in order to be more explicit if it could be both an hex or dec number, and that no "0x" has been prepended.

Reply:
Yeah I think we should make `--base-in` optional